### PR TITLE
refactor(ValidationUtil): add methods accepting error message, deprecate old ones

### PR DESCRIPTION
**Pull request by vursen**
## Description

- Added new methods `validateRequiredConstraint`, `validateMinConstraint` and `validateMaxConstraint` to `ValidationUtil`. They accept a string error message to return in the `ValidationResult` when the check fails.
- Deprecated the previous methods without an error message in favor of the new ones.

Part of https://github.com/vaadin/flow-components/issues/4618

## Type of change

- [x] Refactor

### CODE REVIEW

**Review by DiegoCardoso — APPROVED** [review]

File: vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ValidationUtil.java [hunk 1]
```diff
@@ -28,72 +28,144 @@ private ValidationUtil() {
...
-     * @param minValue
-     *            the min value set on the component
-     * @return <code>Validation.ok()</code> if the validation passes,
-     *         <code>Validation.error()</code> otherwise
+     * @param maxValue
+     *            the maximum value set on the component
+     * @return {@code ValidationResult.ok()} if the value is smaller or equal to
+     *         the maximum, {@code ValidationResult.error()} otherwise
+     */
+    public static <V extends Comparable<V>> ValidationResult checkMax(
```
Comment (yuriy-fix) [inline vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ValidationUtil.java:118]: The new method names could be considered not descriptive enough. Some alternative options to consider:

- `validateMaxConstraint`
- `checkMaxConstraint`
- keep `checkGreaterThanMax`

**Review by yuriy-fix — COMMENTED** [review]
Nit: Consider updating the usage of `ValidationUtil.checkRequired` in other components as well, such as `Checkbox`, `ComboBox`, `RadioButtonGroup`, `Select`, etc.

File: vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ValidationUtil.java (hunk 1 above)
Comment (vursen) [reply vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ValidationUtil.java:118]: Thanks, good point. I like `validateMaxConstraint` the best.

Comment (vursen) [reply vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ValidationUtil.java]: Renamed.

**Comment by vursen** [comment]
> Nit: Consider updating the usage of ValidationUtil.checkRequired in other components as well, such as Checkbox, ComboBox, RadioButtonGroup, Select, etc.

Updated.

**Review by yuriy-fix — APPROVED** [review]

**Pull request merged**